tests/trainer/test_trainer.py

```diff
@@ -157,7 +157,6 @@ def config(self, device: Device, precision: Precision, world_size: int, rank_zer
             'seed': rank_zero_seed,
             'device': device,
             'precision': precision,
-            'deterministic_mode': True,  # testing equivalence
```

Comment:
Since the computed metrics are on state, deterministic mode is needed so the metrics are exactly equal for the checkpoint test. However, since we configure deterministic mode in `conftest.py`, this is no longer needed here.